src/cpp/src/icontinuous_batching.cpp

```diff
@@ -50,7 +50,17 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
         input_ids.reserve(prompts.size());
         timer.start();
         for (const std::string& prompt : prompts) {
-            input_ids.push_back(m_tokenizer.encode(prompt).input_ids);
+            ov::Tensor encoded_inputs;
+            try {
+                ChatHistory history({{{"role", "user"}, {"content", prompt}}});
+                constexpr bool add_generation_prompt = true;
+                auto templated_prompt = m_tokenizer.apply_chat_template(history, add_generation_prompt);
```

Comment:
Looks like you need to fix our test framework to do the same as chat template is applied only for chat cases

The same for tests in .github folder.